frontend/src/components/files/ListingItem.vue

```diff
@@ -247,6 +247,10 @@ export default {
       this.addSelected(this.index);
     },
     open: function () {
+      window.sessionStorage.setItem(
+        this.$route.path,
+        JSON.stringify([document.documentElement.scrollTop, this.index])
+      );
```

Comment:
This is the only right place to save the scrolltop and index to window.sessionStorage before we leave the current list.